platform/pulp/platform/fields.py

```diff
@@ -0,0 +1,80 @@
+"""
+Custom django fields provided by pulp platform
+"""
+import json
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import six, ugettext_lazy as _
+
+
+class JSONField(models.TextField):
+    """
+    A custom django field to serialize dicts into a text field
+
+    """
+    def from_db_value(self, value, expression, connection, context):
+        """
+        Converts a value as returned by the database to a Python object
+
+        :param value: The value to convert to Python
+        :type value: Any
+
+        :param expression:
+        :param connection:
+        :param context:
+
+        :return: A python representation of value
+        :rtype: Any
+        """
+        if isinstance(value, six.string_types):
+            return self.to_python(value)
+
+        return value
+
+    def to_python(self, value):
+        """
+        Converts the value into the correct Python object
+
+        :param value: The value to convert to Python
+        :type value: Any
+
+        :return: A python representation of value
+        :rtype: Any
+        """
+        try:
+            return json.loads(value)
+        except Exception:
+            raise ValidationError(_("Invalid JSON stored in database"))
+
+    def get_db_prep_value(self, value, connection, prepared=False):
+        """
+        Converts value to a backend-specific value
+
+        :param value: The value to be converted
+        :type value: Any
+
+        :param connection:
+
+        :param prepared:
+
+        :return: json string representing the object
+        :rtype: str
+        """
+        if value is None:
+            return None
+
+        return json.dumps(value)
```

Comment:
Is it common practice to document all possible exceptions that could be thrown from dependent code that you didn't write?  I'm more familiar with having to document things you specifically throw, but not the case of all possible exceptions from third parties.